ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PlanNode.java

```diff
@@ -192,10 +195,13 @@ static void throwKeysNotIncludedError(
       final LogicalSchema schema
   ) {
     // When doing a `select *` key columns should be at the front of the column list
-    // but are added at the back during processing for performance reasons.
-    // Switch them around here:
-    final Stream<Column> keys = columns.stream()
-        .filter(c -> schema.isKeyColumn(c.name()));
+    // but are added at the back during processing for performance reasons. Furthermore,
```

Comment:
For my own education. What does this comment mean: `but are added at the back during processing for performance reasons` -- what is the perf impact and why?

Reply:
I think the context here is that it avoids shuffling data (adding at the end of an array is easier than shuffling the whole array)